common.py

```diff
@@ -173,12 +190,6 @@ def from_dict(cls: Type[T], data: Dict[str, Any]) -> T:
         field_type = field.type
```

Comment:
as this function only works with data classes, it would have been a bit nicer to call it `dataclass_from_dict()`. But I see there are a lot of usages already. It's cumbersome to fix now... as you prefer.